### PR TITLE
watchdog: don't boot up VM for 'check_watchdog_support'

### PR DESCRIPTION
**Pull request by PandaWei**
Signed-off-by: Wei Jiangang <weijg.fnst@cn.fujitsu.com>

### CODE REVIEW

**Comment by PandaWei** [comment]
[root@localhost avocado]# scripts/avocado run type_specific.io-github-autotest-qemu.watchdog.watchdog_device_support type_specific.io-github-autotest-qemu.watchdog.boot_with_watchdog
JOB ID     : 37aa229b44588016919c9adedaae8912b509896e
JOB LOG    : /root/avocado/job-results/job-2016-05-13T11.13-37aa229/job.log
TESTS      : 4
 (1/4) type_specific.io-github-autotest-qemu.watchdog.watchdog_device_support.i6300esb: PASS (31.47 s)
 (2/4) type_specific.io-github-autotest-qemu.watchdog.watchdog_device_support.ib700: PASS (33.36 s)
 (3/4) type_specific.io-github-autotest-qemu.watchdog.boot_with_watchdog.i6300esb: PASS (49.16 s)
 (4/4) type_specific.io-github-autotest-qemu.watchdog.boot_with_watchdog.ib700: PASS (45.70 s)
RESULTS    : PASS 4 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2016-05-13T11.13-37aa229/html/results.html
TIME       : 159.69 s

File: qemu/tests/watchdog.py [hunk 1]
```diff
@@ -283,12 +283,14 @@ def hotplug_unplug_watchdog_device():
...
+
+    if test_type != "check_watchdog_support":
+        error.context("Boot VM with WDT(Device:'%s', Action:'%s'), and login"
+                      % (watchdog_device_type, watchdog_action), logging.info)
+        params["start_vm"] = "yes"
+        env_process.preprocess_vm(test, params, env, params.get("main_vm"))
+        vm = env.get_vm(params["main_vm"])
+        session = vm.wait_for_login(timeout=timeout)
 
     if params.get("setup_runlevel") == "yes":
```
Comment (luckyh) [inline qemu/tests/watchdog.py]: I think this if statement section should be included into the fix as well. Others look good to me.

Comment (PandaWei) [inline qemu/tests/watchdog.py]: Thanks for your comments.
Only 'watchdog_action.shutdown_guest' cases declare 'setup_runlevel',  It doesn't mater of ‘watchdog_device_support’.
Of course, Move 'if params.get("setup_runlevel") == "yes"' into 'if test_type != "check_watchdog_support"' block looks better.
I will update it.

**Comment by CongLi** [comment]
Acked-by: Cong Li <coli@redhat.com>

**Comment by luckyh** [comment]
ACK

**Comment by xutian** [comment]
why not update watchdog.cfg, add 'kill_vm_before_test = yes'  'start_vm = no' for check_watchdog_support ?

File: qemu/tests/watchdog.py [hunk 2]
```diff
@@ -283,16 +283,18 @@ def hotplug_unplug_watchdog_device():
...
-                  % (watchdog_device_type, watchdog_action), logging.info)
-    params["start_vm"] = "yes"
-    env_process.preprocess_vm(test, params, env, params.get("main_vm"))
-    vm = env.get_vm(params["main_vm"])
-    session = vm.wait_for_login(timeout=timeout)
-
-    if params.get("setup_runlevel") == "yes":
-        error.context("Setup the runlevel for guest", logging.info)
-        utils_test.qemu.setup_runlevel(params, session)
+
```
Comment (xutian) [inline qemu/tests/watchdog.py]: I think drop line#4 and append below lines in line#18 of watchdog.cfg can fix bug you meet.

Thanks,
Xu

Comment (PandaWei) [inline qemu/tests/watchdog.py]: Modifying the cfg is a good idea. But it doesn't work well.

File: qemu/tests/watchdog.py [hunk 3]
```diff
@@ -283,16 +283,18 @@ def hotplug_unplug_watchdog_device():
...
+    if test_type != "check_watchdog_support":
+        error.context("Boot VM with WDT(Device:'%s', Action:'%s'), and login"
+                      % (watchdog_device_type, watchdog_action), logging.info)
+        params["start_vm"] = "yes"
```
Comment (PandaWei) [inline qemu/tests/watchdog.py]: Here 'start_vm' will be set to 'yes' and boot VM.

Comment (xutian) [inline qemu/tests/watchdog.py]: yes, I think we need to drop this lines.

**Comment by xutian** [comment]
I think check watchdog support should be precondition of other test, it should not be a sub-testcase.

Thanks,
Xu

**Comment by PandaWei** [comment]
Yes, I agree with you.
the watchdog case is added by redhat and adopted by fujitsu.
we think check_watchdog_support with booting VM is unreasonable.
In order avoid any bad influence on redhat,
I add the check "if test_type != "check_watchdog_support"".

If there's no any influence on you,
I would like to remove the check_watchdog_support cases and make it as other case's prerequisite.

**Comment by PandaWei** [comment]
Yes, I agree with you.
the watchdog case is added by redhat and adopted by fujitsu.
we think check_watchdog_support with booting VM is unreasonable.
In order avoid any bad influence on redhat,
I add the check "if test_type != "check_watchdog_support"".

If there's no any influence on you,
I would like to remove the check_watchdog_support cases and make it as other case's prerequisite.

**Comment by xutian** [comment]
there's no any influence on us.

**Comment by PandaWei** [comment]
[root@localhost avocado]# scripts/avocado run type_specific.io-github-autotest-qemu.watchdog.boot_with_watchdog
JOB ID     : 732a88e76f4b31e34b787f4ef8da02016532a695
JOB LOG    : /root/avocado/job-results/job-2016-05-18T12.18-732a88e/job.log
TESTS      : 2
 (1/2) type_specific.io-github-autotest-qemu.watchdog.boot_with_watchdog.i6300esb: PASS (59.82 s)
 (2/2) type_specific.io-github-autotest-qemu.watchdog.boot_with_watchdog.ib700: PASS (62.29 s)
RESULTS    : PASS 2 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0
JOB HTML   : /root/avocado/job-results/job-2016-05-18T12.18-732a88e/html/results.html
TIME       : 122.10 s

**Comment by PandaWei** [comment]
Update

**Comment by xutian** [comment]
Test pass and thanks @PandaWei update. merged.

**Pull request merged**